test/test_log.py

```diff
@@ -1,150 +1,115 @@
+        logging.getLogger().handlers.clear()
+        bigflow.log._LOGGING_CONFIGURED = False
+
+    def test_should_create_correct_logging_link(self):
         with self.assertLogs(level='INFO') as logs:
             # when
-            BigflowLogging.configure_logging('project-id', 'another_logger_name', 'workflow_id')
+            self.configure_mocked_logging('project-id', 'another_log_name', 'workflow_id')
 
         # then
-        self.assertEqual(logs.output, ['INFO:root:\n'
- '*************************LOGS LINK************************* \n'
- ' You can find this workflow logs here: '
- 'https://console.cloud.google.com/logs/query;query=logName%3D%22projects%2Fproject-id%2Flogs%2Fanother_logger_name%22%0Alabels.id%3D%22workflow_id%22 \n'
- '***********************************************************'])
-
-    @mock.patch('bigflow.log.create_logging_client')
-    def test_should_create_correct_logging_link_without_workflow_id(self, create_logging_client_mock):
-        # given
-        create_logging_client_mock.return_value = mock.create_autospec(logging_v2.LoggingServiceV2Client)
-        BigflowLogging.IS_LOGGING_SET = False
+        out = "\n".join(logs.output)
+        self.assertIn("LOGS LINK", out)
+        self.assertIn("https://console.cloud.google.com/logs/query;query=", out)
+        self.assertIn("labels.workflow_id%3D%22workflow_id%22", out)
+
+    def test_should_create_correct_logging_link_without_workflow_id(self):
         with self.assertLogs(level='INFO') as logs:
             # when
-            BigflowLogging.configure_logging('project-id', 'another_logger_name')
+            self.configure_mocked_logging('project-id', 'another_log_name')
 
         # then
-        self.assertEqual(logs.output, ['INFO:root:\n'
- '*************************LOGS LINK************************* \n'
- ' You can find this workflow logs here: '
- 'https://console.cloud.google.com/logs/query;query=logName%3D%22projects%2Fproject-id%2Flogs%2Fanother_logger_name%22%0Alabels.id%3D%22project-id%22 \n'
- '***********************************************************'])
+        out = "\n".join(logs.output)
+        self.assertIn("LOGS LINK", out)
+        self.assertIn("https://console.cloud.google.com/logs/query;query=", out)
+
+    def _assert_single_log_event(self, message_re, severity=None, labels=None):
+        self.assertEqual(1, self.logging_client.write_log_entries.call_count)
+        calls = self.logging_client.write_log_entries.call_args_list[0][0]
+        le = calls[0][0]
+
+        self.assertEqual(le.resource.type, "global")
+        self.assertEqual(le.resource.labels, {"project_id": "project-id"})
+        self.assertEqual(le.log_name, "projects/project-id/logs/logger_name")
+        
+        if severity:
+            self.assertEqual(le.severity, severity)
+        if message_re:
+            self.assertRegexpMatches(le.json_payload['message'], message_re)
+
+        return le
 
     def test_should_log_unhandled_exception(self):
```

Comment:
Imo previous test was better, but this one is simpler.

Reply:
So, should I apply any changes to test?